package trycatch;

import java.util.InputMismatchException;
import java.util.Scanner;

public class TryCatch2 extends Exception{
	
	public TryCatch2 (String name) {
		super(name);
	}
	
	public static void main(String args[]) throws TryCatch2, InputMismatchException {
		try {
		Scanner sc=new Scanner(System.in);
		System.out.println("enter the salary");
		int Salary=sc.nextInt();
		if (Salary>=0 & Salary<=1200) {
			TryCatch2 obj=new TryCatch2("salary is too low");
			throw obj;
		}
		}
		
//		catch(TryCatch2 obj) { //user-defined exception
//			System.out.println(obj.getMessage());
//		}catch(InputMismatchException ime) {
//			System.out.println(ime);	
//		}
//		catch(Exception e) { // pre-defined exceptions
//			e.printStackTrace();
//		}
		
		//2md way: only with one catch-->
		
		
		
		
		catch(Exception e) {
			
			e.printStackTrace();
			
		}
		
		
	}
	
	

}
